compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -1544,6 +1492,7 @@ impl<'v> RootCollector<'_, 'v> {
             ty::ParamEnv::reveal_all(),
             start_def_id,
             self.tcx.mk_args(&[main_ret_ty.into()]),
+            None,
```

Comment:
Using the def span of the start item should always point to the local crate, so it seems better than `None`/`DUMMY_SP`